File: app/src/main/java/joebruckner/lastpick/view/specials/SpecialsComponent.java
package joebruckner.lastpick.view.specials;

import dagger.Subcomponent;
import joebruckner.lastpick.ActivityModule;
import joebruckner.lastpick.ActivityScope;

@ActivityScope
@Subcomponent(modules = {
        ActivityModule.class,
        SpecialsModule.class
        })
public interface SpecialsComponent {
    void inject(SpecialsFragment fragment);
}

File: app/src/main/java/joebruckner/lastpick/view/specials/SpecialsComponent.kt
package joebruckner.lastpick.view.specials

import dagger.Subcomponent
import joebruckner.lastpick.ActivityModule
import joebruckner.lastpick.ActivityScope

@ActivityScope
@Subcomponent(modules = arrayOf(
        ActivityModule::class,
        SpecialsModule::class
))
interface SpecialsComponent {
    fun inject(fragment: SpecialsFragment)
}